### ISSUE TITLE
bug: Strange back button behaviour on android 8 and greater

### ISSUE THREAD

**Issue by divyasri93** [ionic-team/ionic#19971]
# Bug Report

**Ionic version:**

[x] **4.x**

**Current behavior:**
I have ion tabs in the root page of my application. I have a custom backbutton event in app.component.ts which detects the router url and if it matches the tab path, showing an exit popup. 

**Case 1**: First time from the ion-tabs page, on navigating to any inner page and on pressing the hardware back, I have the inner page popped out which is fine but the exit popup is shown simultaneously.

**Case 2**: On navigating to any other tabs and on hitting the back button, I'm taken to the previous tab.

**Expected behavior:**

**Case 1**: After the inner page gets popped out, only on clicking back again, exit popup should be shown.

**Case 2**: App should remain in the same tab, instead of navigating to the previous tab

**Custom back button event handler**

```
   backButtonEvent() {
    this.platform.backButton.subscribeWithPriority(0, async () => {

           const canGoBack = this.routerOutlets
        .toArray()
        .filter(out => out.canGoBack());
      if (canGoBack.length > 0) {
        canGoBack[0].pop();
      } 
        else if (
          this.router.url === "/menu-tabs/menu-tabs" ||
          this.router.url === "/menu-tabs/menu-tabs/home" ||
          this.router.url === "/menu-tabs/menu-tabs/search" ||
          this.router.url === "/menu-tabs/menu-tabs/visits" ||
          this.router.url === "/menu-tabs/menu-tabs/notifications" ||
          this.router.url === "/menu-tabs/menu-tabs/property" ||
        ) {

          if (!this.util.isExitAlertShowing) {
            this.util.isExitAlertShowing = true;
            this.appAlert.showExitApplicationAlert(this.closeAppHandler);
          }
        }
      

    });
  }
```

**Other information:**

Facing this issue mostly in android 8 and greater versions

**Ionic info:**

```
Ionic:

   Ionic CLI                     : 5.4.6 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.2.0
   @angular-devkit/build-angular : 0.13.8
   @angular-devkit/schematics    : 7.2.4
   @angular/cli                  : 7.3.8
   @ionic/angular-toolkit        : 1.4.1

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.0.0, browser 5.0.4
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 3.1.2, (and 28 other plugins)

Utility:

   cordova-res : not installed
   native-run  : 0.2.9 

System:

   Android SDK Tools : 26.1.1 (/opt/adt-bundle-linux-x86_64-20140702/sdk/)
   NodeJS            : v10.15.3 (/usr/local/bin/node)
   npm               : 6.4.1
   OS                : Linux 4.4
```

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you try this on the latest version of Ionic? The info you posted says you are running `@ionic/angular@4.2.0`, and the latest is `@ionic/angular@4.11.5`.

**Comment by divyasri93** [comment]
@liamdebeasi I've upgraded to the latest angular version as mentioned.

**Here is the ionic info:**

Ionic:

   Ionic CLI                     : 5.4.6 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.11.5
   @angular-devkit/build-angular : 0.13.8
   @angular-devkit/schematics    : 7.2.4
   @angular/cli                  : 7.3.8
   @ionic/angular-toolkit        : 1.4.1

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.0.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 3.1.2, (and 28 other plugins)

Utility:

   cordova-res : not installed
   native-run  : 0.2.9 

System:

   Android SDK Tools : 26.1.1 (/opt/adt-bundle-linux-x86_64-20140702/sdk/)
   NodeJS            : v10.15.3 (/usr/local/bin/node)
   npm               : 6.4.1
   OS                : Linux 4.4


The issue that I had mentioned in **Case 1** has been fixed and works fine. 
But the issue in **Case 2** remains as is. It occurs randomly. Once it occurs, the following happens:

- When a popup is closed using hardware back, it closes the popup and the page behind it as well. 
- Even after setting a page as root page, on clicking back, other pages remains in the stack without getting cleared.
- Always navigates to previous tab on back press
